WeakAuras/RegionTypes/Icon.lua

```diff
@@ -525,9 +527,10 @@ local function modify(parent, region, data)
     end
 
     function region:SetTime(duration, expirationTime)
-      if (duration > 0) then
+      local adjustedDuration = region.progressScale or duration;
+      if (adjustedDuration > 0) then
         cooldown:Show();
-        cooldown:SetCooldown(expirationTime - duration, duration);
+        cooldown:SetCooldown(expirationTime - adjustedDuration, adjustedDuration);
```

Comment:
Did you test that? I'm not sure the cooldown can handle start times that are in the future. E.g. if adjustedDuration is less than duration.